bin/commands/indexDump.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const
+  fs = require('fs'),
+  ColorOutput = require('./colorOutput'),
+  path = require('path'),
+  ndjson = require('ndjson'),
+  getSdk = require('./getSdk');
+
+//@TODO use 'recursive: true' with node.js 10
+function mkdirp (fullPath) {
+  const parts = fullPath.split(path.sep);
+
+  if (parts.length > 1) {
+    mkdirp(parts.slice(0, parts.length - 1).join('/'));
+  }
+
+  if (! fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+function writeData(stream, data) {
+  return new Promise(resolve => stream.write(data, () => resolve()));
```

Comment:
To be tested, but that might be very slow, as this is like waiting synchronously for each `data` line to be handled by the system.
You may want to discard that callback argument when invoking `write` and instead use the returned value: you can resolve the promise immediately if it returns `true`, otherwise this means that backpressure is building up and you need to wait for the `drain` event before resolving the promise.

This might not be important in this case, and I didn't test the performances difference, so this is your call.